bynder_sdk/client/upload_client.py

```diff
@@ -1,144 +1,127 @@
+        :param file_name: The name of the file to be uploaded.
+        :param file_size: The size of the file to be uploaded.
+        :param chunks_count: The number of chunks in which the file was
+        uploaded.
         """
-        return self.session.post(
-            '/v4/upload/{0}/'.format(init_data['s3file']['uploadid']),
+        self.session.post(
+            '/v7/file_cmds/upload/{}/finalise_api'.format(file_id),
             data={
-                'id': init_data['s3file']['uploadid'],
-                'targetid': init_data['s3file']['targetid'],
-                's3_filename': init_data['s3_filename'],
-                'chunks': total_parts
+                'fileName': file_name,
+                'fileSize': file_size,
+                'chunksCount': chunks_count,
+                "sha256": self.file_sha256,
+                "intent": "upload_main_uploader_asset",
             }
         )
 
-    def _save_media(self, import_id, data, media_id=None):
-        """ Saves a new media asset in Bynder. If media id is specified
-        in the query a new version of the asset will be saved. Otherwise
-        a new asset will be saved.
-        """
-        poll_status = self._poll_status(import_id)
-        if import_id not in poll_status['itemsDone']:
-            raise Exception('Converting media failed')
-
-        save_endpoint = '/v4/media/save/{}/'.format(import_id)
-        if media_id:
-            save_endpoint = '/v4/media/{}/save/{}/'.format(
-                media_id, import_id)
-            data = {}
-
-        return self.session.post(
-            save_endpoint,
-            data=data
-        )
-
-    def _poll_status(self, import_id):
-        """ Gets poll processing status of finalised files.
+    def _save_media(self, file_id, data, media_id=None):
+        """Saves the completely uploaded file.
+        :param file_id: The uuid4 used to identify the file to be uploaded.
+        :param data: The upload_data containing asset information.
+        :param media_id: The media_id of the asset to be created or updated.
+        :return: - success: boolean that indicate the result of the upload
+        call.
+                - mediaitems: a list of mediaitems created, with at least
+                the original.
+                - batchId: the batchId of the upload.
+                - mediaId: the mediaId update or created.
         """
-        for _ in range(MAX_POLLING_ITERATIONS):
-            time.sleep(POLLING_IDLE_TIME)
-            status_dict = self.session.get(
-                '/v4/upload/poll/',
-                params={'items': [import_id]}
-            )
-
-            if [v for k, v in status_dict.items() if import_id in v]:
-                return status_dict
-
-        # Max polling iterations reached => upload failed
-        status_dict['itemsFailed'].append(import_id)
-        return status_dict
+        if data['brandId'] and data['brandId'].strip():
```

Comment:
brandId (or a body in general) is only required for the case where no media ID is passed. And I wonder why the `data['brandId'].strip()` check is there?

Reply:
the `strip()` check is there to make sure the client doesn't pass something like `'   '` to just escape the required `brandId`. So instead of hitting our endpoints and get a response that the `brandId` is invalid, we just do it here before even hitting the endpoint.

This is just a sanity check tbh.

Reply:
Hmm sounds a bit arbitrary to me though why just for this field. Besides, in case it would have spaces in it, we would still pass that as part of the data. So if we want to do stuff like stripping spaces, I'd rather do that as part of some validating step in general (but I wouldn't recommend doing that in this PR and maybe it would be a bit overkill in general anyway).

Reply:
Alright I will drop the `strip()` 👍